src/core/net/SRP_CODER.md

```diff
@@ -0,0 +1,353 @@
+   | C |  int value        |         | C |        7-bit value        |
+   +---+---+---+---+---+---+         +---+---+---+---+---+---+---+---+
+```
+
+Similar integer coding schemes ([Variable-Length Quantity (VLQ)](https://en.wikipedia.org/wiki/Variable-length_quantity)) are used in other protocols.
+
+### Examples
+
+Assuming the first segment is one byte (8-bit) long:
+
+- Numbers less than (2^7) 128 are encoded as a single byte.
+- Numbers between 128 and (2^14 - 1) 16,383 are encoded as two bytes.
+- Numbers between 16,384 and (2^21 - 1) 2,097,151 are encoded as three bytes.
+
+Value `4460 = 0x1234` is encoded as two bytes: `[0xa4, 0x34]`
+
+```
+                             (0x1234)
+  +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+  | 0   0   0   1   0   0   1   0 | 0   0   1   1   0   1   0   0 |
+  +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+          /         7-bit            / \         7-bit             \
+         /                          /   \                           \
+        /                          /     \                           \
+   +---+---+---+---+---+---+---+---+  +---+---+---+---+---+---+---+---+
+   | 1 | 0   1   0   0   1   0   0 |  | 0 | 0   1   1   0   1   0   0 |
+   +---+---+---+---+---+---+---+---+  +---+---+---+---+---+---+---+---+
+              (0xa4)                             (0x34)
+```
+
+Assuming the first segment is 6-bit long:
+
+- Numbers less than (2^5) 32 are encoded in the first segment.
+- Numbers between 32 and (2^12 - 1) 4,095 use two segments (the first 6-bit segment and an additional byte).
+
+## DNS Name and Label Format
+
+### Label Dispatch
+
+```
+    0   1   2   3   4   5   6   7
+  +---+---+---+---+---+---+---+---+
+  | Type  ... | ...               |
+  +---+---+---+---+---+---+---+---+
+
+```
+
+- `00 <6-bit len>`: Standard label encoding, where the lower 6 bits indicate the label's length (number of characters). This is followed by the label characters, totaling `len` in number.
+- `01 <6-bit len>`: Similar to the above, but the label begins with an underscore `'_'`. The `len` value here doesn't include the underscore itself, which isn't explicitly encoded after the dispatch byte.
+- `10 <6-bit offset first seg>`: This references a previously encoded label within the message. The 6-bit value in the dispatch byte serves as the first segment of a compact `uint` representing the offset to that label. This offset must point to the label dispatch byte of the previous label.
+  - Note: This mechanism applies to a single label only. Any subsequent labels are encoded directly after the current one. This differs from traditional DNS pointer name compression, where a pointer indicates that the remaining labels should be read from the specified offset.
+- `110 <5-bit code>`: Commonly used constant labels. Codes are:
+  - `0`: `_udp`
+  - `1`: `_tcp`
+  - `2`: `_matter`
+  - `3`: `_matterc`
+  - `4`: `_matterd`
+  - `5`: `_hap`
```

Comment:
What about _sub? This is used for subtypes, which I think are common with Matter.

Reply:
> What about _sub? This is used for subtypes, which I think are common with Matter.

The subtype names are already compressed. Check the "Add Service Block" section.

There is no need to encode or include the `_sub` label in the coded message. The coding scheme aims to omit known/fixed information to compress the message, and the `_sub` label is a good example of this.